### PR TITLE
Music: support flyout toolbox

### PR DESCRIPTION
**Pull request by breville**
This adds support for an optional Music Lab level property, `"type": "flyout"`, which will show a [toolbox](https://developers.google.com/blockly/guides/configure/web/toolbox) with all blocks shown all the time, without categories.  

It seems this could be helpful in early levels in a progression, when only a few blocks will be available.

The toolbox section of the level data has been changed a little, to have separate `"blocks"` and optional `"type"` entries.

### flyout toolbox

Screenshot 2023-07-21 at 12 43 38 AM

### categories toolbox (default)

Screenshot 2023-07-21 at 12 43 43 AM

### CODE REVIEW

File: dashboard/test/ui/features/star_labs/musiclab/musiclab_drag_block.feature [hunk 1]
```diff
@@ -27,5 +27,5 @@ Scenario Outline: Dragging play sound block
 
 Examples:
   | url                                                       | test_name               |
-  | http://studio.code.org/s/allthethings/lessons/46/levels/2 | music lab script level  |
+  | http://studio.code.org/s/allthethings/lessons/46/levels/4 | music lab script level  |
```
Comment (sanchitmalhotra126) [inline dashboard/test/ui/features/star_labs/musiclab/musiclab_drag_block.feature:30]: Should we change this to a standalone project URL (projects/music/new) so we don't need to worry about updating it if the allthethings progression changes?

File: dashboard/config/scripts/levels/Music Level 1.level [hunk 2]
```diff
@@ -9,6 +9,7 @@
       "toolbox": {
         "Play": ["play_sound_at_current_location_simple2"]
       },
+      "simpleToolbox": true,
```
Comment (sanchitmalhotra126) [inline dashboard/config/scripts/levels/Music Level 1.level]: Minor: should we nest this inside the "toolbox" field? Something like
```
"toolbox": {
   "blocks": { "Play": [...] },
   "simple": true,
   // alternatively
   "type": "flyout" / "category"
}
```

**Review by sanchitmalhotra126 — APPROVED** [review]
A few minor things but LGTM! Nice addition!

File: dashboard/config/scripts/levels/Music Level 1.level (hunk 2 above)
Comment (breville) [reply dashboard/config/scripts/levels/Music Level 1.level]: Good idea.  I was opting for legacy compatibility, but we're still at the stage where we should build the ideal format, and this is closer to that!

File: dashboard/test/ui/features/star_labs/musiclab/musiclab_drag_block.feature (hunk 1 above)
Comment (breville) [reply dashboard/test/ui/features/star_labs/musiclab/musiclab_drag_block.feature:30]: We depend on `allthethings` for much of our UI testing, so I think it's reasonable to keep these particular levels in sync with UI tests.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**